### PR TITLE
Use external directive-handler library

### PR DESCRIPTION
**Pull request by alexarchambault**
This makes Scala CLI use directive handler classes from [an external project](https://github.com/scala-cli/directive-handler), rather than from its sources. This allows other projects to handle using directives with the same high level API as Scala CLI, rather than the rather low level [using_directives](https://github.com/VirtusLab/using_directives). I'm trying to [use this library from Almond](https://github.com/almond-sh/almond/pull/1124), for example.

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -461,7 +461,8 @@ trait Directives extends ScalaCliSbtModule with ScalaCliPublishModule with HasTe
     Deps.bloopConfig,
     Deps.jsoniterCore,
     Deps.pprint,
-    Deps.usingDirectives
+    Deps.usingDirectives,
+    ivy"sh.almond::directive-handler:0.13.14-33-77bfc73b-SNAPSHOT"
```
Comment (tgodzik) [inline build.sc]: Ideally this would be under the same org as the normal using directives, no?

Comment (alexarchambault) [reply build.sc]: Yeah, my bad, that's stale code. Just changed that.

**Comment by alexarchambault** [comment]
I'd be ok giving you access to [the directive-handler repo](https://github.com/scala-cli/directive-handler), or having you fork it under the VirtusLab org, and manage it from there.

**Comment by tgodzik** [comment]
> I'd be ok giving you access to [the directive-handler repo](https://github.com/scala-cli/directive-handler), or having you fork it under the VirtusLab org, and manage it from there.

I think it would be best to fork it under VirtusLab since it's a core part of ScalaCLI and it would be easier for use manage access for people in the team

**Comment by tgodzik** [comment]
@romanowski could you fork the repository?

**Comment by romanowski** [comment]
Sure: https://github.com/VirtusLab/directive-handler

**Comment by tgodzik** [comment]
@alexarchambault where do we stand with this? Could we instead publish the library from the usual Scala CLI repo?

I think that's what we discussed at the call, or am I totally mistaken?

**Comment by Gedochao** [comment]
Closing this, there'll be another approach to this soon.
To be tracked under #2443.

**Pull request closed without merging**